edgelet/provisioning/src/provisioning.rs

```diff
@@ -836,12 +850,15 @@ where
                     move |mut prov_result| {
                         debug!("Provisioning result {:?}", prov_result);
                         let reconfigure = match prov_result.reconfigure {
-                            ReprovisioningStatus::DeviceDataUpdated => {
```

Comment:
So DPS will not return DeviceDataUpdated anymore or edge will not handle this substatus?

Reply:
It was added speculatively by Anca. DPS never returned `DeviceDataUpdated`, and we never had any code to parse it from DPS's response it even if it did, so the code path was never used.